digdag-core/src/main/java/io/digdag/core/database/DatabaseProjectStoreManager.java

```diff
@@ -693,11 +735,18 @@ public StoredProjectMapper(ConfigMapper cfm)
         public StoredProject map(int index, ResultSet r, StatementContext ctx)
                 throws SQLException
         {
+            String name = r.getString("name");
+            Optional<Instant> deletedAt = Optional.absent();
+            if (r.wasNull()) {
```

Comment:
In my model I would use `name` to show "the input users gave when the user starts the session.". I.e., the `name` is permanent and `key` is ephemeral. `name` is the project name that the user specified. `key` is assigned the value of `name` when the project is pushed.

To use a filesystem analogue, `key` is like a directory entry pointing to an inode. When the directory entry is deleted/removed/unlinked, the `key` is removed/released and can be reused to refer to another inode (project).

It feels like we're a bit into bike shedding territory here though. Both of our approaches are mostly isomorphic and I feel it's more important to merge and release this feature than to achieve perfect consensus.